### PR TITLE
FH-3626 - Change the cmd  auth to follow the V3 standard and add unit tests

### PR DESCRIPTION
**Pull request by camilamacedo86**
JIRA: https://issues.jboss.org/browse/FH-3626

**Changes**
- Move to V3 standard 
- Add unit test

**Local Test**

( the same behavior for before and after the change as follows )

Before
```
Camilas-MacBook-Pro:fh-fhc cmacedo$ fhc auth x46d64qpkqecaxn7qqelmfle 6bwkef6snekytmej3ggdelqg C7B25DA8EA544A948A5094BC0CF4BDD8 {}
fhc ERR! Error in auth call: unknown_policyId
fhc Command executed with error
Camilas-MacBook-Pro:fh-fhc cmacedo$ 
```
```

Camilas-MacBook-Pro:fh-fhc cmacedo$ fhc auth x46d64qpkqecaxn7qqelmfle eb76818ea4e26305901101afa34bff56715a8e9f C7B25DA8EA544A948A5094BC0CF4BDD8 {}
fhc ERR! Error: Error - not 2xx status code. (400)
fhc ERR! {"date":"Sun Aug 20 11:56:59 UTC 2017 - 1503230219837","msg":"invalid guid: [eb76818ea4e26305901101afa34bff56715a8e9f], length is not 24 chars","status":"error"}
fhc ERR! Request ID:  70ed8836-f758-4c2c-8fea-a51e5e89d20b
fhc ERR! 
fhc ERR! System Darwin 15.6.0
fhc ERR! command "/Users/cmacedo/.nvm/versions/node/v4.4.3/bin/node" "/Users/cmacedo/.nvm/versions/node/v4.4.3/bin/fhc" "auth" "x46d64qpkqecaxn7qqelmfle" "eb76818ea4e26305901101afa34bff56715a8e9f" "C7B25DA8EA544A948A5094BC0CF4BDD8" "{}"
fhc Command executed with error
```

After
```
Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js auth x46d64qpkqecaxn7qqelmfle 6bwkef6snekytmej3ggdelqg C7B25DA8EA544A948A5094BC0CF4BDD8 {}
fhc ERR! Error in auth call: unknown_policyId
fhc Command executed with error
```
```

Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js auth x46d64qpkqecaxn7qqelmfle eb76818ea4e26305901101afa34bff56715a8e9f C7B25DA8EA544A948A5094BC0CF4BDD8 {}
fhc ERR! Error: Error - not 2xx status code. (400)
fhc ERR! {"date":"Sun Aug 20 11:57:15 UTC 2017 - 1503230235037","msg":"invalid guid: [eb76818ea4e26305901101afa34bff56715a8e9f], length is not 24 chars","status":"error"}
fhc ERR! Request ID:  dece9f21-895f-4750-ad95-1226ac2c6c0c
fhc ERR! 
fhc ERR! System Darwin 15.6.0
fhc ERR! command "/Users/cmacedo/.nvm/versions/node/v4.4.3/bin/node" "/Users/cmacedo/work/fh-fhc/bin/fhc.js" "auth" "x46d64qpkqecaxn7qqelmfle" "eb76818ea4e26305901101afa34bff56715a8e9f" "C7B25DA8EA544A948A5094BC0CF4BDD8" "{}"
fhc Command executed with error


```

### CODE REVIEW

**Comment by camilamacedo86** [comment]
please retest it

File: lib/cmd/fh3/auth.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/* globals i18n */
+var fhreq = require("../../utils/request");
+var common = require("../../common.js");
+
+module.exports = {
+  'desc' : i18n._('Performs an auth call against an Auth Policy'),
+  'examples' : [{
+    cmd : 'fhc auth --policy=<policy> --app=<app> --device=<device> --params=<params>',
+    desc : i18n._('Performs a auth call against  Auth Policy')}],
+  'demand' : ['policy', 'app', 'device', 'params'],
```
Comment (wtrocki) [inline lib/cmd/fh3/auth.js]: are `params` required? Should be?

File: lib/cmd/fh3/auth.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    0 : 'policy',
+    1 : 'app',
+    2 : 'device',
+    3 : 'params'
+  },
+  'describe' : {
+    'policy' : i18n._("Unique 24 character GUID of your Auth Policy."),
+    'app' : i18n._("Unique 24 character GUID of your client application."),
+    'device' : i18n._("Unique 24 character GUID of the device"),
+    'params' : i18n._("The auth parameters specific to the auth policy")
```
Comment (wtrocki) [inline lib/cmd/fh3/auth.js]: Can you provide more info here. 
Both device id and params need to be taken from ngui. 
I do not see why someone will want to use this command.

File: lib/cmd/fh3/auth.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  },
+  'customCmd': function(argv, cb) {
+    if (typeof argv.params === 'string') {
+      argv.params = JSON.parse(argv.params);
+    }
+    common.doApiCall(fhreq.getFeedHenryUrl(), "/box/srv/1.1/admin/authpolicy/auth", {
+      policyId: argv.policy,
+      clientToken: argv.app,
+      device: argv.device,
+      params: JSON.stringify(argv.params)
```
Comment (wtrocki) [inline lib/cmd/fh3/auth.js]: Antipattern - Passing json as command line tool.

File: lib/cmd/fh3/auth.js (hunk 1 above)
Comment (camilamacedo86) [reply lib/cmd/fh3/auth.js]: This is the way that it was before. I think so. What do you think? Is it making sense?

File: lib/cmd/fh3/auth.js (hunk 2 above)
Comment (camilamacedo86) [reply lib/cmd/fh3/auth.js]: What information do you think that we could provide? Do you think we should remove this command?

File: lib/cmd/fh3/auth.js (hunk 3 above)
Comment (camilamacedo86) [reply lib/cmd/fh3/auth.js]: I will perform this change.

**Comment by camilamacedo86** [comment]
@wtrocki Tks for your review. So, I saw that you approve I just would like to confirm with you before performing the merge. The change request was made. Please, could you check and tell me if you think that is ok to be merged or if we should do something more here?

**Pull request merged**